### PR TITLE
Token API Code Documentation

### PR DESCRIPTION
**Pull request by adecaro**
This PR adds comments to the Token API. In also remove some unused methods, structs, and interfaces.

Signed-off-by: Angelo De Caro <adc@zurich.ibm.com>

### CODE REVIEW

File: token/metadata.go [hunk 1]
```diff
@@ -37,6 +40,7 @@ func (m *Metadata) GetToken(raw []byte) (*token2.Token, view.Identity, []byte, e
 	return tok, id, tokenInfoRaw, nil
 }
 
+// SpentTokenID returns the token IDs of the tokens that ware spent by the Token Request this metadata is associated.
```
Comment (KElkhiyaoui) [inline token/metadata.go:43]: that ware spent by the Token Request this metadata is associated with.

File: token/provider.go [hunk 2]
```diff
@@ -14,27 +14,40 @@ var (
...
 
+// VaultProvider provides token vault instances
 type VaultProvider interface {
+	// Vault returns a token vault instance for the passed inputs
 	Vault(network string, channel string, namespace string) tokenapi.Vault
 }
 
+// SelectorManager provides instances of Selector
 type SelectorManager interface {
+	// NewSelector returns a new Selector instance bound the passed id
```
Comment (KElkhiyaoui) [inline token/provider.go:31]: bound to the passed id.

File: token/provider.go [hunk 3]
```diff
@@ -14,27 +14,40 @@ var (
...
 	NewSelector(id string) (Selector, error)
-	Unlock(txID string) error
+	// Unlock unlocks the tokens bound to the passed if, if any
```
Comment (KElkhiyaoui) [inline token/provider.go:33]: to the passed id,

File: token/certification.go [hunk 4]
```diff
@@ -3,37 +3,47 @@ Copyright IBM Corp. All Rights Reserved.
...
 package token
 
 import (
 	"github.com/hyperledger-labs/fabric-token-sdk/token/driver"
 	token2 "github.com/hyperledger-labs/fabric-token-sdk/token/token"
 )
 
+// CertificationManager is the interface for managing token certification as described by the paper
+// [`Privacy-preserving auditable token payments in a permissioned blockchain system`]('https://eprint.iacr.org/2019/1058.pdf')
 type CertificationManager struct {
```
Comment (mbrandenburger) [inline token/certification.go:16]: CertificationManager is a struct, not an interface. Does it implement some interface?

File: token/certification.go [hunk 5]
```diff
@@ -3,37 +3,47 @@ Copyright IBM Corp. All Rights Reserved.
...
 func (c *CertificationManager) Certify(wallet *CertifierWallet, ids []*token2.ID, tokens [][]byte, request []byte) ([][]byte, error) {
 	return c.c.Certify(wallet.w, ids, tokens, request)
 }
 
+// VerifyCertifications verfiies the certification of the passed token ids.
 func (c *CertificationManager) VerifyCertifications(ids []*token2.ID, certifications [][]byte) error {
 	return c.c.VerifyCertifications(ids, certifications)
 }
 
+// CertificationClient is the interface for the client side of the certification
```
Comment (mbrandenburger) [inline token/certification.go:36]: interface?

File: token/metadata.go (hunk 1 above)
Comment (adecaro) [reply token/metadata.go]: done

File: token/provider.go (hunk 2 above)
Comment (adecaro) [reply token/provider.go]: done

File: token/provider.go (hunk 3 above)
Comment (adecaro) [reply token/provider.go]: done

File: token/provider.go [hunk 6]
```diff
@@ -14,27 +14,40 @@ var (
...
 }
 
+// VaultProvider provides token vault instances
 type VaultProvider interface {
+	// Vault returns a token vault instance for the passed inputs
 	Vault(network string, channel string, namespace string) tokenapi.Vault
 }
 
+// SelectorManager provides instances of Selector
 type SelectorManager interface {
```
[the remark this answers is not in the document]
Comment (adecaro) [reply token/provider.go]: done

File: token/certification.go (hunk 4 above)
Comment (adecaro) [reply token/certification.go]: done

File: token/certification.go (hunk 5 above)
Comment (adecaro) [reply token/certification.go]: done

File: token/certification.go [hunk 7]
```diff
@@ -3,37 +3,47 @@ Copyright IBM Corp. All Rights Reserved.
...
 type CertificationClient struct {
 	cc driver.CertificationClient
 }
 
+// IsCertified returns true if the passed token id has been already certified
```
[the remark this answers is not in the document]
Comment (adecaro) [reply token/certification.go]: done

File: token/provider.go [hunk 8]
```diff
@@ -14,27 +14,40 @@ var (
...
-	Unlock(txID string) error
+	// Unlock unlocks the tokens bound to the passed if, if any
+	Unlock(id string) error
 }
 
+// SelectorManagerProvider provides instances of SelectorManager
 type SelectorManagerProvider interface {
+	// SelectorManager returns a new SelectorManager instance for the passed inputs.
+	// If the selector manager has already been created, it returns the existing instance.
 	SelectorManager(network string, channel string, namespace string) SelectorManager
```
[the remark this answers is not in the document]
Comment (adecaro) [reply token/provider.go]: done

File: token/vault.go [hunk 9]
```diff
@@ -26,6 +29,7 @@ func (q *QueryEngine) UnspentTokensIterator() (*UnspentTokensIterator, error) {
 	return &UnspentTokensIterator{UnspentTokensIterator: it}, nil
 }
 
+// UnspentTokensIteratorBy is an iterator over all unspent tokens in this vault owned by passed id and whose token type matches the passed token type
 func (q *QueryEngine) UnspentTokensIteratorBy(id, typ string) (*UnspentTokensIterator, error) {
 	it, err := q.qe.UnspentTokensIteratorBy(id, typ)
 	if err != nil {
```
[the remark this answers is not in the document]
Comment (adecaro) [reply token/vault.go]: done

**Pull request merged**